pkg/reconciler/revision/background.go

```diff
@@ -180,7 +178,7 @@ func (r *backgroundResolver) addWorkItems(rev *v1.Revision, name types.Namespace
 	for i := range rev.Spec.Containers {
 		image := rev.Spec.Containers[i].Image
 
-		r.queue.Add(&workItem{
+		r.queue.AddRateLimited(&workItem{
```

Comment:
It might be too early and I might need a :coffee: but: Why does this even work? We're passing a pointer to `workItem` and it doesn't seem to implement some `Equals` interface or whatever, but still the deduplication **seems** to work? :thinking:

Reply:
It doesn't, that's part of the follow up I was talking about to make per-item back off work (also not calling Forget after errors). This PR fixes global rate limits only.

Reply:
ahhhhh, that makes sense, thanks for the clarification!